Tasks/SonarQubePreBuild/SonarQubeHelper.ps1

```diff
@@ -0,0 +1,230 @@
+###############################################################################################
+### Note: This file should be kept in sync with the SonarQubeHelper.ps1 from the "end task"####
+###############################################################################################
+
+
+# When passing arguments to a process, the quotes need to be doubled and   
+# the entire string needs to be placed inside quotes to avoid issues with spaces  
+function EscapeArg  
+{  
+    param([string]$argVal)  
+  
+    $argVal = $argVal.Replace('"', '""');  
+    $argVal = '"' + $argVal + '"';  
+  
+    return $argVal;  
+}  
+
+
+# Set a variable in a property bag that is accessible by all steps
+# To retrieve the variable use $val = Get-Variable $distributedTaskContext "varName"
+function SetTaskContextVariable
+{
+    param([string][ValidateNotNullOrEmpty()]$varName, 
+          [string]$varValue)
+    
+    Write-Host "##vso[task.setvariable variable=$varName;]$varValue"
+}
+
+function GetTaskContextVariable()
+{
+    param([string][ValidateNotNullOrEmpty()]$varName)
+    return Get-TaskVariable -Context $distributedTaskContext -Name $varName
+}
+
+#
+# Helper that informs if a "filePath" has been specified. The platform will return the root of the repo / workspace if the user enters nothing.
+#
+function IsFilePathSpecified
+{
+     param([string]$path)
+
+     if ([String]::IsNullOrWhiteSpace($path))
+     {
+        return $false
+     }
+
+     return ![String]::Equals(
+                [System.IO.Path]::GetFullPath($path).TrimEnd([System.IO.Path]::DirectorySeparatorChar, [System.IO.Path]::AltDirectorySeparatorChar),
+                [System.IO.Path]::GetFullPath($env:BUILD_SOURCESDIRECTORY).TrimEnd([System.IO.Path]::DirectorySeparatorChar, [System.IO.Path]::AltDirectorySeparatorChar),
+                [StringComparison]::OrdinalIgnoreCase)
+}
+
+
+#
+# Retry a specific command until it completes without exceptions. Usage:  
```

Comment:
"Throws if the command didn't complete successfully after the specified number of retries."